app/refund/refund.go

```diff
@@ -23,10 +24,18 @@ func NewGasRefundHandler(ak auth.AccountKeeper, sk types.SupplyKeeper, ik innert
 	) (refundFee sdk.Coins, err error) {
 		var gasRefundHandler sdk.GasRefundHandler
 
-		if tx.GetType() == sdk.EvmTxType {
-			gasRefundHandler = evmGasRefundHandler
+		if tmtypes.HigherThanEarth(ctx.BlockHeight()) {
+			if tx.GetType() == sdk.EvmTxType || tx.GetType() == sdk.StdTxType {
```

Comment:
No need to check tx type here, because tx type must be EvmTxType or StdTxType. Suggest to refactor the whole NewGasRefundHandler function.